stdlib/kokorawr/kokorawr/functions/services/ttt.js

```diff
@@ -43,70 +39,89 @@ function isTie(marked, gameOver) {
-		return callback(null, {text: "Can't reset: game not over\n" + markedToBoard(marked), success: false})
-	} else if(lastTeam == team) {
-		let color = 'Red';
-	    if(team != 0) {
-	      color = 'Blue';
-	    }
-		return callback(null, {text: `It is not ${color}'s turn.`, success: false})
-	} else if (marked[loc-1] != '~'){
-		return callback(null, {text: `Location ${loc} already marked!\n` + markedToBoard(marked), success: false});
-	} else {
-		marked[loc-1] = team;
-		lastTeam = team;
+	lib.utils.storage.get('ttt', (err, ttt) => {
+		if (err) {
+			return callback(null, 'An error has occurred with your command');
+		}
+		if (ttt == null) {
+			ttt = {};
+		}
+		if (!ttt.hasOwnProperty('marked')) {
+			ttt['marked'] = ['~', '~', '~', '~', '~', '~', '~', '~', '~'];
+		}
+		if (!ttt.hasOwnProperty('gameOver')) {
+			ttt['gameOver'] = false;
+		}
+		if (!ttt.hasOwnProperty('lastTeam')) {
+			ttt['lastTeam'] = Math.round(Math.random());
+		}
+		let marked = ttt['marked'];
+		let gameOver = ttt['gameOver'];
+		let lastTeam = ttt['lastTeam'];
 
-		marked[0] == marked[1] && marked[1] == marked[2] && marked[1] != '~' ? gameOver = true: "N/A";
-		marked[3] == marked[4] && marked[4] == marked[5] && marked[4] != '~' ? gameOver = true: "N/A";
-		marked[6] == marked[7] && marked[7] == marked[8] && marked[7] != '~' ? gameOver = true: "N/A";
+		if(turn) {
+		    let color = 'Red';
+		    if(lastTeam == 0) {
+		      color = 'Blue';
+		    }
+		    return callback(null, {text: `It is ${color}'s turn!`, success: true});
+	  	} else if(state) {
+	  		return callback(null, {text: '\n'+markedToBoard(marked), success: true});
+	  	} else if(reset && gameOver) {
+			ttt = {};
+			let color = 'Red';
+		    if(lastTeam == 0) {
+		      color = 'Blue';
+		    }
+		    lib.utils.storage.set('ttt', ttt, (err, ttt) => {
+		    	return callback(null, {text: `Successfully Reset! It is ${color}'s turn!`, success: true});
+		    });
+		} else if(gameOver) {
+			return callback(null, {text: 'Game already over\n' + markedToBoard(marked), success: false})
+		} else if(reset) {
+			return callback(null, {text: "Can't reset: game not over\n" + markedToBoard(marked), success: false})
+		} else if(lastTeam == team) {
+			let color = 'Red';
+		    if(team != 0) {
+		      color = 'Blue';
+		    }
+			return callback(null, {text: `It is not ${color}'s turn.`, success: false})
+		} else if (marked[loc-1] != '~'){
+			return callback(null, {text: `Location ${loc} already marked!\n` + markedToBoard(marked), success: false});
+		} else {
+			ttt['marked'][loc-1] = team;
+			ttt['lastTeam'] = team;
 
-		marked[0] == marked[3] && marked[3] == marked[6] && marked[3] != '~' ? gameOver = true: "N/A";
-		marked[1] == marked[4] && marked[4] == marked[7] && marked[4] != '~' ? gameOver = true: "N/A";
-		marked[2] == marked[5] && marked[5] == marked[8] && marked[5] != '~' ? gameOver = true: "N/A";
+			marked[0] == marked[1] && marked[1] == marked[2] && marked[1] != '~' ? gameOver = true: "N/A";
```

Comment:
I wish we could clean up these lines, but not sure how... haha

Reply:
We could potentially create an array containing arrays of locations that should be equal [[1, 2, 3], [4, 5, 6], etc] and have a for loop parse through that.

Reply:
Yeah I think that sounds cleaner!